src/include/duckdb/common/file_system.hpp

```diff
@@ -75,6 +76,24 @@ class FileFlags {
 	static constexpr uint8_t FILE_FLAGS_APPEND = 1 << 5;
 };
 
+class FileType {
```

Comment:
Can this be an enum class?

Reply:
Hmm, probably. I just wanted a portable layer between what `Win` and `*nix` supports, as POSIX is not cutting it...